### PR TITLE
[folly related] update to 10.07

### PR DESCRIPTION
**Pull request by c8ef**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by dg0yt** [comment]
> ~~~
> -- Could NOT find PythonInterp (missing: PYTHON_EXECUTABLE)

The failing port is `fbthrift`. But that port doesn't take any efforts to deal with python. Something like:
~~~cmake
vcpkg_find_acquire_program(PYTHON3)

vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
    OPTIONS
        "-DPYTHON_EXECUTABLE=${PYTHON3}"
)
~~~

The other triplets might find a system python (linux, osx), or port `python3` is installed for other reasons, early enough.

**Comment by dg0yt** [comment]
I can't say much more but I see the top-level CMakeLists.txt still has the deprecated `find_package(PythonInterp)` while the failing script has `find_package(Python3 COMPONENTS Interpreter)`. Maybe a bad mix?

**Comment by MonicaLiu0311** [comment]
Except `libaio & liburing`, other features are  tested successfully in the following triplet:
```
x64-windows
x64-windows-static
```

**Review by MonicaLiu0311 — DISMISSED** [review]

File: ports/mvfst/fix-cmake.patch [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+From: Carlo Cabrera <30379873+carlocab@users.noreply.github.com>
+Date: Tue, 8 Oct 2024 04:09:34 +0800
+Subject: [PATCH 1/2] Add `QuicTransportBaseLite` to CMake build
+
+This is the analogue of 0b2743fdae9b746659815afdf00611fe7999282e for
+CMake.
+---
+ quic/api/CMakeLists.txt | 1 +
+ 1 file changed, 1 insertion(+)
+
```
Comment (MonicaLiu0311) [inline ports/mvfst/fix-cmake.patch:11]:
```suggestion
```

File: ports/mvfst/fix-cmake.patch [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+From: Carlo Cabrera <30379873+carlocab@users.noreply.github.com>
+Date: Tue, 8 Oct 2024 04:24:34 +0800
+Subject: [PATCH 2/2] Add `AckedPacketIterator` to CMake build
+
+This is the analogue of 654c5b90d2e9431e71f0dd3d5be990200306acc4 for
+CMake.
+---
+ quic/state/CMakeLists.txt | 1 +
+ 1 file changed, 1 insertion(+)
+
```
Comment (MonicaLiu0311) [inline ports/mvfst/fix-cmake.patch:35]:
```suggestion
```

Comment (MonicaLiu0311) [reply ports/mvfst/fix-cmake.patch:35]: To minimize the size of the patch, could you please remove this information?

**Review by MonicaLiu0311 — APPROVED** [review]

**Review by data-queue — APPROVED** [review]

**Pull request merged**